vulnerabilities/importers/rockylinux.py

```diff
@@ -0,0 +1,199 @@
+from vulnerabilities.importer import VulnerabilitySeverity
+from vulnerabilities.rpm_utils import rpm_to_purl
+from vulnerabilities.utils import get_cwe_id
+from vulnerabilities.utils import get_item
+from vulnerabilities.utils import requests_with_5xx_retry
+
+logger = logging.getLogger(__name__)
+
+# FIXME: we should use a centralized retry
+requests_session = requests_with_5xx_retry(max_retries=5, backoff_factor=1)
+
+
+def fetch_cves() -> Iterable[List[Dict]]:
+    page_no = 0
+    cve_data_list = []
+    while True:
+        current_url = f"https://errata.rockylinux.org/api/v2/advisories?filters.product=&filters.fetchRelated=true&page={page_no}&limit=100"
+        try:
+            response = requests_session.get(current_url)
+            if response.status_code != requests.codes.ok:
+                logger.error(f"Failed to fetch RedHat CVE results from {current_url}")
+                break
+            cve_data = response.json().get("advisories") or []
+            cve_data_list.extend(cve_data)
+        except Exception as e:
+            logger.error(f"Failed to fetch rockylinux CVE results from {current_url} {e}")
+            break
+        if not cve_data:
+            break
+        page_no += 1
+    return cve_data_list
+
+
+class RockyLinuxImporter(Importer):
+    spdx_license_expression = "CC-BY-4.0"
+    license_url = "https://access.redhat.com/security/data"
+    importer_name = "Rocky Importer"
+
+    def advisory_data(self) -> Iterable[AdvisoryData]:
+
+        for rockylinux_cve in fetch_cves():
+            yield to_advisory(rockylinux_cve)
+
+
+def to_advisory(advisory_data):
+    aliases = advisory_data.get("name") or ""
+    date_published = dateparser.parse(advisory_data.get("publishedAt", ""))
+
+    summary = advisory_data.get("description") or ""
+    affected_products = advisory_data.get("affectedProducts") or []
+    affected_packages = []
+    for products in affected_products:
+        packages = advisory_data["rpms"][products]["nvras"]
+        affected_packages.extend(packages)
+    processed_affected_packages: List[AffectedPackage] = []
+    for rpm in affected_packages:
+        purl = rpm_to_purl(rpm_string=rpm.rsplit(".rpm", 1)[0] or "", namespace="rocky-linux")
+        if purl:
+            try:
+                affected_version_range = RpmVersionRange.from_versions(sequence=[purl.version])
+                processed_affected_packages.append(
+                    AffectedPackage(
+                        package=PackageURL(
+                            type=purl.type,
+                            name=purl.name,
+                            namespace=purl.namespace,
+                            qualifiers=purl.qualifiers,
+                            subpath=purl.subpath,
+                        ),
+                        affected_version_range=affected_version_range,
+                        fixed_version=None,
+                    )
+                )
+            except Exception as e:
```

Comment:
@ambuj-1211 Please avoid using a general exception and use the appropriate exception for this case instead.